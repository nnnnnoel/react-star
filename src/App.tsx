import * as React from "react";
import StarRate from "./StarRate";

function App() {
  const [value, setValue] = React.useState<number>(0);

  const onChange = React.useCallback((rate: number) => {
    setValue(rate);
  }, []);

  return <StarRate {...{ onChange, value }} />;
}

export default App;
