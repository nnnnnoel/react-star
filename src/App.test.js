import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

test("renders successfully at root", () => {
  const elem = render(<App />);
  expect(elem).toMatchSnapshot();
});
