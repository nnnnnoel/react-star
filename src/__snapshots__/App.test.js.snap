// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully at root 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app"
      >
        <div
          class="container"
        >
          <svg
            class="pointer left"
            height="32"
            width="32"
          >
            star-empty.svg
          </svg>
          <svg
            class="pointer left"
            height="32"
            width="32"
          >
            star-empty.svg
          </svg>
          <svg
            class="pointer left"
            height="32"
            width="32"
          >
            star-empty.svg
          </svg>
          <svg
            class="pointer left"
            height="32"
            width="32"
          >
            star-empty.svg
          </svg>
          <svg
            class="pointer left"
            height="32"
            width="32"
          >
            star-empty.svg
          </svg>
        </div>
        <svg
          class="pointer more-left"
          height="16"
          width="16"
        >
          reset.svg
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app"
    >
      <div
        class="container"
      >
        <svg
          class="pointer left"
          height="32"
          width="32"
        >
          star-empty.svg
        </svg>
        <svg
          class="pointer left"
          height="32"
          width="32"
        >
          star-empty.svg
        </svg>
        <svg
          class="pointer left"
          height="32"
          width="32"
        >
          star-empty.svg
        </svg>
        <svg
          class="pointer left"
          height="32"
          width="32"
        >
          star-empty.svg
        </svg>
        <svg
          class="pointer left"
          height="32"
          width="32"
        >
          star-empty.svg
        </svg>
      </div>
      <svg
        class="pointer more-left"
        height="16"
        width="16"
      >
        reset.svg
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
